### ISSUE TITLE
False positive `invalid-argument-type`: `Literal[True]` not assignable to `**kwargs: Any`

### ISSUE THREAD

**Issue by arikon** [astral-sh/ty#3169]
### Summary

When calling a function with `**kwargs: t.Any`, passing a keyword argument with a `Literal[True]` value (e.g. `is_flag=True`) produces a false positive `invalid-argument-type` error saying `Expected Any, found Literal[True]`.

`Literal[True]` should be assignable to `Any`.

### Reproduction

```python
import typing as t

def option(*args: str, **kwargs: t.Any) -> t.Callable:
    ...

@option("--debug", is_flag=True, help="Enable debug")
def cli(debug: bool):
    pass
```

```
error[invalid-argument-type]: Argument to function `option` is incorrect
  |
5 | @option("--debug", is_flag=True, help="Enable debug")
  |                    ^^^^^^^^^^^^ Expected `Any`, found `Literal[True]`
  |
```

### Real-world example

This triggers on virtually every `click.option()` call, since `click.option` is defined as:

```python
def option(*param_decls: str, cls: type[Option] | None = None, **attrs: t.Any) -> t.Callable[[FC], FC]:
```

### Expected behavior

`Literal[True]` (and any `Literal` type) should be assignable to `Any` without errors.

### Version

```
ty 0.0.26
```

**Comment by charliermarsh** [comment]
Are you able to put together a minimally reproducible example in our playground? The snippet above doesn't reproduce for me: https://play.ty.dev/597b3749-0cab-40cd-a8e1-cf575d2cc99f

**Comment by sharkdp** [comment]
Thank you for reporting this.

I can not reproduce this (see [Playground](https://play.ty.dev/280184a3-4a3a-4092-b68d-a4489ecd428d)). Is there anything specific to your configuration that might be important?

**Comment by JP01** [comment]
I'm seeing this too but I cannot recreate in the playground.

uvx ty check /path/to/file.py

```
error[invalid-argument-type]: Argument to function `__new__` is incorrect

raise Exception("test")
                 ^^^^ Expected `Any`, found `Literal["test"]`

info: Function defined here
    --> stdlib\builtins.pyi:4549:9
     |
4547 |     __traceback__: TracebackType | None
4548 |     def __init__(self, *args: object) -> None: ...
4549 |     def __new__(cls, *args: Any, **kwds: Any) -> Self: ...
     |         ^^^^^^^      ---------- Parameter declared here
4550 |     def __setstate__(self, state: dict[str, Any] | None, /) -> None: ...
4551 |     def with_traceback(self, tb: TracebackType | None, /) -> Self:
     |
info: Union variant `def __new__(cls, *args: Any, **kwds: Any) -> Unknown` is incompatible with this call site
info: Attempted to call union type `(def __new__(cls, *args: Any, **kwds: Any) -> Unknown) | (bound method Exception.__init__(*args: object) -> None)`
info: rule `invalid-argument-type` is enabled by default
```

But on the playground that doesn't seem to cause the error.

I'm on python311, windows.

**Comment by carljm** [comment]
@JP01 Do you have a minimal repro for it that you can provide in full (all files, all contents)? Does the minimal repro from the OP reproduce the issue for you locally?

@arikon I assume the minimal repro in your OP does reproduce for you locally? Does it reproduce if placed in a file in a directory by itself, with no `pyproject.toml` or `ty.toml` in any ancestor directory? What's your OS?

**Comment by JP01** [comment]
Hey! 
Sorry I should have been more specific.
I am getting this on a single python file in my `documents/src` folder (but also happens anywhere else)

The builtins.pyi it's using seem to be coming from the vendored `typeshed` in vscode at least... 
But that wouldn't explain why I'm also getting it when I just run it directly via `uvx` :(

I don't have any custom personal typeshed to my knowledge.

It would suggest to me that `Any` isn't properly being evaluated to accept Any type?

Windows 11
python311
ty 0.0.29 (438a78d68 2026-04-05)

```

def doit():
    raise Exception("test")

if __name__ == "__main__":
    doit()
```

<img width="876" height="219" alt="Image" src="https://github.com/user-attachments/assets/3e12764f-fa7a-46e3-835b-d13c374078a3" />

<img width="1014" height="916" alt="Image" src="https://github.com/user-attachments/assets/ca82035d-3d19-4e1d-82d3-149932db10ff" />

**Comment by carljm** [comment]
Well this is very odd; no explanation comes to mind. To be clear, if you create an empty directory, where there is no `ty.toml` or `pyproject.toml` in any ancestor directory, and put a `main.py` file in that directory containing just this:

```py
def doit():
    raise Exception("test")

if __name__ == "__main__":
    doit()
```

Then `uvx ty@0.0.29 check main.py` in that directory reproduces this error for you?

Do you have any `PYTHONPATH` environment variable set?
